homeassistant/components/sensor/emoncms.py

```diff
@@ -0,0 +1,253 @@
+"""
+Support for monitoring emoncms feeds.
+
+https://emoncms.org
+"""
+from datetime import timedelta
+import logging
+import requests
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util as util
+import homeassistant.util.dt as dt_util
+from homeassistant.helpers.event import track_point_in_utc_time
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_API_KEY, CONF_URL, CONF_VALUE_TEMPLATE,
+    CONF_UNIT_OF_MEASUREMENT, CONF_ID, CONF_SCAN_INTERVAL,
+    STATE_UNKNOWN)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers import template
+
+
+_LOGGER = logging.getLogger(__name__)
+
+DECIMALS = 2
+CONF_EXCLUDE_FEEDID = "exclude_feed_id"
+CONF_INCLUDE_FEEDID = "include_feed_id"
+CONF_INCLUDE_FEEDID_NAMES = "include_feed_id_names"
+DEFAULT_SCAN_INTERVAL = 60
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_URL): cv.string,
+    vol.Required(CONF_ID): cv.string,
+    vol.Optional(CONF_EXCLUDE_FEEDID, default=[]):
+        vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_INCLUDE_FEEDID, default=[]):
+        vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_INCLUDE_FEEDID_NAMES, default=[]):
+        vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
+    vol.Optional(CONF_UNIT_OF_MEASUREMENT, default="W"): cv.string,
+    vol.Optional(CONF_SCAN_INTERVAL):
+        vol.All(vol.Coerce(int), vol.Range(min=5)),
+
+})
+
+ATTR_SIZE = 'Size'
+ATTR_LASTUPDATETIME = 'LastUpdated'
+ATTR_TAG = 'Tag'
+ATTR_FEEDID = 'FeedId'
+ATTR_USERID = 'UserId'
+ATTR_FEEDNAME = 'FeedName'
+ATTR_LASTUPDATETIMESTR = 'LastUpdatedStr'
+
+
+def get_id(sensorid, feedtag, feedname, feedid, feeduserid):
+    """Return unique identifier for feed / sensor."""
+    return "emoncms" + sensorid + "_" + feedtag + "_" + \
+           feedname + "_" + feedid + "_" + feeduserid
+
+
+# pylint: disable=too-many-locals
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Emoncms sensor."""
+    apikey = config.get(CONF_API_KEY)
+    url = config.get(CONF_URL)
+    sensorid = config.get(CONF_ID)
+    value_template = config.get(CONF_VALUE_TEMPLATE)
+    unit_of_measurement = config.get(CONF_UNIT_OF_MEASUREMENT)
+    exclude_feeds = config.get(CONF_EXCLUDE_FEEDID)
+    include_feeds = config.get(CONF_INCLUDE_FEEDID)
+    include_feeds_names = config.get(CONF_INCLUDE_FEEDID_NAMES)
+    interval = util.convert(config.get(CONF_SCAN_INTERVAL), int,
+                            DEFAULT_SCAN_INTERVAL)
+
+    include_length = len(include_feeds)
+    exclude_length = len(exclude_feeds)
+    include_names_length = len(include_feeds_names)
+
+    if include_length > 0 and exclude_length > 0:
```

Comment:
You can check this with voluptuous as well: 
```python
a=vol.Schema({
    vol.Exclusive('include', 'in_out_group'): str, 
    vol.Exclusive('exclude', 'in_out_group'): str
})
```

and possibly `cv.has_at_least_one_key` if you need at least one